### PR TITLE
Show subtitle text color setting

### PR DESCRIPTION
**Pull request by Madh93**
**Changes**
Show text color option in subtitle appearance setting:

imagen

**Issues**
N/A

### CODE REVIEW

**Comment by Madh93** [comment]
> > I have always wondered why this was hidden... man_shrugging
> 
> Probably because you need to know what to write in it.
> 
> I prefer this field to be replaced by the color picker. Btw, there is option for the background color.

Yeah, by default the value for the background color is `transparent`, but if you specify some color the result is kind of weird:

imagen
imagen

**Comment by dmitrylyzo** [comment]
> Yeah, by default the value for the background color is `transparent`, but if you specify some color the result is kind of weird:

I know. This is what they look like in Firefox.
Appearance is slightly better in Chrome:
subs-1

Maybe we should replace it with background opacity?

I'm not talking about doing it immediately - just to think about it.

**Comment by dmitrylyzo** [comment]
As an alternative to the color picker, the text field can be replaced with the select.
Palette? I think we can start with just `white` and `yellow`. At most, we can use a full set of web colors.

**Comment by alexladeira** [comment]
Hi, just a jellyfin user here. I know is not very polite to ask for ETA, but... Can we have someone looking into this?

**Comment by Madh93** [comment]
My apologies for the slow reply @dmitrylyzo @thornbill 

I've replaced the text field by a basic HTML5 color picker. It works great but I don't like how it's show.

Firefox:
imagen

Chrome:
imagen

**Review by dmitrylyzo — COMMENTED** [review]
Much better.

In webOS 1.2 it displays correctly, but the color selection doesn't work. We could show `actionSheet` with a list of predefined colors.

**Comment by Madh93** [comment]
@dmitrylyzo I've added an `actionSheet` with a list of predefined colors that it must be shown in webOS and related devices. In any other case, browsers will show the color picker.

WebOS 6.0:

webos
webos2

Firefox:

browser

File: src/components/subtitlesettings/subtitlesettings.js [hunk 1]
```diff
@@ -201,6 +204,10 @@ function embed(options, self) {
                 sliderVerticalPosition.classList.add('focusable');
                 sliderVerticalPosition.enableKeyboardDragging();
             }, 0);
+
+            // Replace color picker
+            options.element.querySelector('#inputTextColor').closest('div').classList.add('hide');
+            options.element.querySelector('#selectTextColor').closest('div').classList.remove('hide');
```
Comment (dmitrylyzo) [inline src/components/subtitlesettings/subtitlesettings.js:210]: no `closest` in webOS 1.2
```suggestion
            dom.parentWithTag(options.element.querySelector('#inputTextColor'), 'DIV').classList.add('hide');
            dom.parentWithTag(options.element.querySelector('#selectTextColor'), 'DIV').classList.remove('hide');
```

**Review by dmitrylyzo — COMMENTED** [review]
I was thinking to listen the `click` event and show `actionSheet`, but your way also fine.

My suggestions (other than `closest`) probably require discussion.

File: src/components/subtitlesettings/subtitleappearancehelper.js [hunk 2]
```diff
@@ -62,9 +62,34 @@ function getTextStyles(settings, preview) {
...
 
-    const textColor = settings.textColor || '#ffffff';
-    if (textColor) {
-        list.push({ name: 'color', value: textColor });
+    const hexTextColor = settings.hexTextColor || '#ffffff';
+
+    switch (settings.textColor || 'white') {
+        case 'yellow':
+            list.push({ name: 'color', value: '#ffff00' });
+            break;
```
[the remark this answers is not in the document]
Comment (Madh93) [reply src/components/subtitlesettings/subtitleappearancehelper.js:70]: Nice approach I didn't know :man_facepalming:

File: src/components/subtitlesettings/subtitlesettings.js (hunk 1 above)
Comment (Madh93) [reply src/components/subtitlesettings/subtitlesettings.js:210]: Oh thank you! I tried to install the webOs 1.2 emulator with no luck so I using the webOS 6.0 simulator instead.

**Comment by garret** [comment]
Sorry to ask the obvious but has this PR been accepted into Jellyfin stable? 
I am running the latest docker container image from linuxserver and have also the latest android app version but can't see the possibility to choose a color for the subtitles 🙈

**Comment by nielsvanvelzen** [comment]
This PR is not in 10.8.x, it will be released with 10.9.

**Comment by bfg100k** [comment]
> This PR is not in 10.8.x, it will be released with 10.9.

Any idea when 10.9 will drop and/or why this PR cannot be included in 10.8.x? Being able to change the subtitle color globally will make a big difference to those HDR videos!

**Comment by nielsvanvelzen** [comment]
Only bugfixes go into current releases, everything else will go in the next version. We don't have an ETA, could be a few weeks or multiple months.

**Comment by YoungbloodTheOne** [comment]
please don't forget gray color (#888888) in the list, for oled TVs , this avoid screen burn

**Comment by hajosattila** [comment]
> please don't forget gray color (#888888) in the list, for oled TVs , this avoid screen burn

+1

**Pull request merged**